roscpp_azure_iothub/src/ros_azure_iothub_cpp_node.cpp

```diff
@@ -216,10 +218,56 @@ void sendMsgToAzureIoTHub(const char* msg, IOTHUB_DEVICE_CLIENT_HANDLE deviceHan
     messagecount++;
 }
 
+// Message serialization for sending topic messages via reported properties 
+static char* serializeToJson(std::string topic, std::string message)
+{
+    char* result = NULL;
+
+    JSON_Value* root_value = json_value_init_object();
+    JSON_Object* root_object = json_value_get_object(root_value);
+
+    std::string topic_header = "ros_messages." + topic;
+    std::string msg = "[" + message + "]";
+    (void)json_object_dotset_value(root_object, topic_header.c_str(), json_parse_string(msg.c_str()));
+
+    result = json_serialize_to_string_pretty(root_value);
+    puts(result);
+
+    json_value_free(root_value);
+
+    return result;
+}
+
+static void reportedStateCallback(int status_code, void* userContextCallback)
+{
+    (void)userContextCallback;
+    printf("Device Twin reported properties update completed with result: %d\r\n", status_code);
+}
+
+void buildReportedString(std::string& topic_msg, std::string msg, bool& first)
+{
+    if(first)
```

Comment:
Could you use topic_msg.empty() to determine if you need to add a comma instead of maintaining the first variable? (Less code to maintain)